### PR TITLE
Java Grep and JDBC Apps

### PR DESCRIPTION
**Pull request by FatimaAnwer**

### CODE REVIEW

File: core_java/README.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Core Java Apps
+This project is under development. Since this project follows the GitFlow, the final work will be merged to the master branch after Team Code Team.
```
Comment (jarviscanada) [inline core_java/README.md]: remove this if you are done

File: core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+        JavaGrepImp javaGrepImp = new JavaGrepImp();
+        javaGrepImp.setRegex(args[0]);
+        javaGrepImp.setRootPath(args[1]);
+        javaGrepImp.setOutFile(args[2]);
+
+        try{
+            javaGrepImp.process();
+        }catch (Exception ex){
+            javaGrepImp.logger.error(ex.getMessage(), ex);
```
Comment (jarviscanada) [inline core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java]: javaGrepImp.logger.error(~ex.getMessage()~ "meaningful message", ex)

File: core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+
+            while((line = reader.readLine()) != null){
+                allLines.add(line);
+            }
+
+            reader.close();
+            return allLines;
+
+        }catch (FileNotFoundException e){
+            throw new IllegalArgumentException("File does not exist");
```
Comment (jarviscanada) [inline core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java]: throw new IllegalArgumentException("File does not exist", e);

you will lose the stack trace if you don't pass it to the new exception. 

fix this for all occurrences

File: core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepLambdaImp.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+            JavaGrepLambdaImp javaGrepLambdaImp = new JavaGrepLambdaImp();
+            javaGrepLambdaImp.setRegex(args[0]);
+            javaGrepLambdaImp.setRootPath(args[1]);
+            javaGrepLambdaImp.setOutFile(args[2]);
+
+            try {
+                javaGrepLambdaImp.process();
+            } catch (Exception ex) {
+                ex.printStackTrace();
```
Comment (jarviscanada) [inline core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepLambdaImp.java]: use logger instead. never use printstacktrace

File: core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepLambdaImp.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    @Override
+    public List<String> readLines(File inputFile) throws IllegalArgumentException {
+
+        try{
+            Path path = Paths.get(inputFile.getAbsolutePath());
+            return Files.lines(path).collect(Collectors.toList());
+        }catch (IOException e){
+            logger.error("Lines from the file cannot be read", e);
+        }
+        return null;
```
Comment (jarviscanada) [inline core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepLambdaImp.java]: **Path path = null;**
        try{
            **path = Paths.get(inputFile.getAbsolutePath());**
            return Files.lines(path).collect(Collectors.toList());
        }catch (IOException e){
            logger.error("Lines from the file cannot be read", e);
        }
        **return path;**

File: core_java/grep/src/main/java/ca/jrvs/apps/practice/HelloWorld.java [hunk 6]
```diff
@@ -0,0 +1,10 @@
+package ca.jrvs.apps.practice;
+
+class HelloWorld {
+
+    // Your program begins with a call to main().
+    // Prints "Hello, World" to the terminal window.
+    public static void main(String args[]) {
```
Comment (jarviscanada) [inline core_java/grep/src/main/java/ca/jrvs/apps/practice/HelloWorld.java]: remove

File: core_java/jdbc/src/main/java/ca/jrvs/apps/jdbc/CustomerDAO.java [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+                customer.setLastName(resultSet.getString("last_name"));
+                customer.setEmail(resultSet.getString("email"));
+                customer.setPhone(resultSet.getString("phone"));
+                customer.setAddress(resultSet.getString("address"));
+                customer.setCity(resultSet.getString("city"));
+                customer.setState(resultSet.getString("state"));
+                customer.setZipCode(resultSet.getString("zipcode"));
+            }
+        }catch(SQLException e){
+            e.printStackTrace();
```
Comment (jarviscanada) [inline core_java/jdbc/src/main/java/ca/jrvs/apps/jdbc/CustomerDAO.java]: never printstacktrack, use logger instead. please fix for all occurrences

File: practice/target/maven-status/maven-compiler-plugin/compile/default-compile/inputFiles.lst [hunk 8]
```diff
@@ -0,0 +1 @@
+/Users/usman/Desktop/dev/jarvis_data_eng_FatimaAnwer/practice/src/main/java/ca/jrvs/practice/dataStructure/HelloWorld.java
```
Comment (jarviscanada) [inline practice/target/maven-status/maven-compiler-plugin/compile/default-compile/inputFiles.lst:1]: git rm -r  practice/target

**Review by jarviscanada — APPROVED** [review]
LGTM.

**Comment by FatimaAnwer** [comment]
comments addressed

**Pull request merged**